internal/server/server_test.go

```diff
@@ -0,0 +1,190 @@
+	"time"
+
+	"github.com/charmbracelet/log"
+	"github.com/dicedb/dice/config"
+	"github.com/dicedb/dice/internal/clientio"
+	"github.com/dicedb/dice/internal/server"
+	"github.com/dicedb/dice/testutils"
+)
+
+func TestAbortCommand(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Cleanup(cancel)
+
+	var wg sync.WaitGroup
+	runTestServer(ctx, &wg)
+
+	time.Sleep(2 * time.Second)
+
+	// Test 1: Ensure the server is running
+	t.Run("ServerIsRunning", func(t *testing.T) {
+		conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", config.Port))
+		if err != nil {
+			t.Fatalf("Failed to connect to server: %v", err)
+		}
+		conn.Close()
+	})
+
+	//Test 2: Send ABORT command and check if the server shuts down
+	t.Run("AbortCommandShutdown", func(t *testing.T) {
+		conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", config.Port))
+		if err != nil {
+			t.Fatalf("Failed to connect to server: %v", err)
+		}
+		defer conn.Close()
+
+		// Send ABORT command
+		result := fireCommand(conn, "ABORT")
+		if result != "OK" {
+			t.Fatalf("Unexpected response to ABORT command: %v", result)
+		}
+
+		// Wait for the server to shut down
+		time.Sleep(1 * time.Second)
+
+		// Try to connect again, it should fail
+		_, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", config.Port))
+		if err == nil {
+			t.Fatal("Server did not shut down as expected")
+		}
+	})
+
+	// Test 3: Ensure the server port is released
+	t.Run("PortIsReleased", func(t *testing.T) {
+		// Try to bind to the same port
+		listener, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", config.Port))
+		if err != nil {
+			t.Fatalf("Port should be available after server shutdown: %v", err)
+		}
+		listener.Close()
+	})
+
+	wg.Wait()
+}
+
+func TestServerRestartAfterAbort(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	// start test server.
+	var wg sync.WaitGroup
+	runTestServer(ctx, &wg)
+
+	time.Sleep(1 * time.Second)
+
+	conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", config.Port))
+	if err != nil {
+		t.Fatalf("Server should be running after restart: %v", err)
+	}
+
+	// Send ABORT command to shut down server
+	result := fireCommand(conn, "ABORT")
+	if result != "OK" {
+		t.Fatalf("Unexpected response to ABORT command: %v", result)
+	}
+	conn.Close()
+
+	// wait for the server to shutdown
+	time.Sleep(2 * time.Second)
+
+	// restart server
+	runTestServer(ctx, &wg)
+
+	// wait for the server to start up
+	time.Sleep(2 * time.Second)
+
+	// Check if the server is running
+	conn, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", config.Port))
+	if err != nil {
+		t.Fatalf("Server should be running after restart: %v", err)
+	}
+
+	// Clean up
+	result = fireCommand(conn, "ABORT")
+	if result != "OK" {
+		t.Fatalf("Unexpected response to ABORT command: %v", result)
+	}
+	conn.Close()
+
+	wg.Wait()
+}
+
+//nolint:unused
+func runTestServer(ctx context.Context, wg *sync.WaitGroup) {
```

Comment:
We can export the runTestServer and fireCommand methods from the package in which they are defined to avoid code duplication.

Reply:
Okay cool. I will work on this.

Reply:
I have fixed this already.